Source/Optional.swift

```diff
@@ -1,3 +1,17 @@
+public func <<-<T,U>(f: T? -> U, a: T?) -> U? {
+    switch a {
+    case .Some(_): return .Some(f(a))
+    case .None: return .None
+    }
```

Comment:
If we were to pull this in, I'd want to follow the precedent we set with the other operators and implement `extend` in an extension on `Optional` and have this call `a.extend(f)`.